tx/textual/valuerenderer/message.go

```diff
@@ -0,0 +1,79 @@
+package valuerenderer
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"sort"
+	"strings"
+
+	"google.golang.org/protobuf/reflect/protoreflect"
+)
+
+type messageValueRenderer struct {
+	tr *Textual
+}
+
+var _ ValueRenderer = (*messageValueRenderer)(nil)
+
+func (mr *messageValueRenderer) Format(ctx context.Context, v protoreflect.Value, w io.Writer) error {
+	fields := v.Message().Descriptor().Fields()
+	fds := make([]protoreflect.FieldDescriptor, 0, fields.Len())
+	for i := 0; i < fields.Len(); i++ {
+		fds = append(fds, fields.Get(i))
+	}
+	sort.Slice(fds, func(i, j int) bool { return fds[i].Number() < fds[j].Number() })
+
+	fmt.Fprintf(w, "%s object\n", v.Message().Descriptor().Name())
+	buf := &bytes.Buffer{}
```

Comment:
Nit - consider moving `buf` to be declared where it's first used. The runtime savings of reusing the Buffer should be negligible.